graylog2-server/src/main/java/org/graylog2/lookup/caches/CaffeineLookupCache.java

```diff
@@ -0,0 +1,277 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.lookup.caches;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.Expiry;
+import com.github.benmanes.caffeine.cache.stats.CacheStats;
+import com.github.benmanes.caffeine.cache.stats.StatsCounter;
+import com.google.auto.value.AutoValue;
+import com.google.inject.Inject;
+import com.google.inject.assistedinject.Assisted;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.graylog.autovalue.WithBeanGetter;
+import org.graylog2.plugin.lookup.LookupCache;
+import org.graylog2.plugin.lookup.LookupCacheConfiguration;
+import org.graylog2.plugin.lookup.LookupCacheKey;
+import org.graylog2.plugin.lookup.LookupResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import javax.inject.Named;
+import javax.validation.constraints.Min;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class CaffeineLookupCache extends LookupCache {
```

Comment:
Would it be possible to replace the usage of the guava cache in `GuavaLookupCache` with caffeine? The caffeine cache can do everything that the guava cache can do, right? That way we don't have to explain to users which cache to use in which situations.

The internal naming in the code wouldn't be accurate anymore, but I guess that's not a huge issue. We can even rename it as long as we don't change the `NAME` value.

What do you think?

Reply:
We could do that. Making Caffeine optional gives us the relieve that we don't break anything, though ;-)

Reply:
As discussed offline, let's migrate the existing cache over to caffeine. :smile: